dali/operators/decoder/audio/generic_decoder.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_AUDIO_GENERIC_DECODER_H_
+#define DALI_OPERATORS_DECODER_AUDIO_GENERIC_DECODER_H_
+
+#include <sndfile.h>
+#include <cassert>
+#include <cstring>
+#include <memory>
+#include <vector>
+#include "dali/core/format.h"
+#include "dali/core/error_handling.h"
+#include "dali/operators/decoder/audio/audio_decoder.h"
+
+
+namespace dali {
+
+/**
+ * Generic decoder, that serves as a fallback to most of the formats we need.
+ * It uses `libsnd` for decoding.
```

Comment:
nitpick: this info doesn't need to be here (it's an implementation detail)

Reply:
Removed